dspace-api/src/main/java/org/dspace/content/authority/Concept.java

```diff
@@ -445,10 +446,16 @@ public static Concept find(Context context, int id) throws SQLException
      */
 
     public static ArrayList<Concept> findByJournalID(Context context, String journalID)
-            throws SQLException
+            throws SQLException, AuthorizeException
     {
         ArrayList<Concept> concepts = new ArrayList<Concept>();
-        TableRowIterator row = DatabaseManager.query(context,"select c.* from concept as c, conceptmetadatavalue as cmv where cmv.text_value = '"+journalID+"' and cmv.parent_id=c.id;");
+        String schema = "journal";
+        String element = "journalID";
+        MetadataSchema mds = MetadataSchema.find(context, schema);
+        MetadataField mdf = MetadataField.findByElement(context, mds.getSchemaID(), element, null);
+        journal_field_id = mdf.getFieldID();
+        log.info ("journal field id is " + journal_field_id);
+        TableRowIterator row = DatabaseManager.query(context, "select c.* from concept as c, conceptmetadatavalue as cmv where upper(cmv.text_value) = ? and cmv.parent_id = c.id and cmv.field_id = ?;", journalID, journal_field_id);
```

Comment:
This assumes that journalID is uppercase, but there isn't code in this method to ensure it is uppercase.